### ISSUE TITLE
gam-sae: the installed-clamp derivative `priced_clamp_adjoint_extras` exists only on the dense exact-A route; the matrix-free lane has no counterpart and it is unverified whether it needs one

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#2915]
Split from #2333, where this was recorded as an unverified candidate only in the thread ([comment of 2026-09-11 21:41](https://github.com/SauersML/gam/issues/2333), section 5).

At `origin/main`, `priced_clamp_adjoint_extras` chains the priced basin's diagonal clamp derivative to the strength and theta coordinates. Its callers are `dense_exact_a_logdet_channels` (production, dense exact-A route) and `exact_a_theta_adjoint_joint` (`#[cfg(test)] mod test_support`). The matrix-free lane prices clamp-basin rows through `row_deflation_is_live` but has no explicit `dE/dtheta` leg.

**Question.** Does the matrix-free outer gradient omit a term the dense exact-A route carries on a fit whose exact-A factor records clamp-basin rows? Answer by comparing the two routes' outer gradients on one such state, or by a central difference of the matrix-free criterion value.

**Comment by SauersML** [comment]
## Answer by reading, at `04561e22c`: yes, the matrix-free lane omits the clamp price's own derivative

None of this has been run. It is a derivation from the code cited below. The acceptance at the end is the measurement that would confirm it.

### What the lane's value prices

`factor_spectral_deflated_criterion_row_with_geometry` classifies each eigendirection `v_a` of a row's raw exact-A block (`crates/gam-solve/src/arrow_schur/factorization.rs:631`). `classify_exact_a_direction` (`solve_options.rs:55`) prices a clamp-basin direction at

```text
λ̃_a = λ_a + c_a,     c_a = v_aᵀ E v_a        (solve_options.rs:68)
```

`E` is the row's `clamp_diag`, i.e. `materialize_ard_concave_clamp_diagonal_for_rows`. The installed row block is `Φ = Σ_a λ̃_a v_a v_aᵀ`. The direction keeps `RowSpectralConditioning::Raw`, and only `cond_evals` records the new price.

### What the lane's gradient differentiates

The from-probes channels (`assignment_log_strength_hessian_trace_from_probes`, `construction_quasi_laplace.rs:5184`; `logdet_theta_adjoint_from_probes`, `:6439`) contract the reconstructed row-block selected inverse `G` against the Daleckii–Krein operator `DΦ[dA] = U(F ⊙ UᵀdAU)Uᵀ`. `F` comes from `row_deflation_frechet_coefficients` (`:5541`): `F_ab = (λ̃_a − λ̃_b)/(λ_a − λ_b)` off the diagonal, and `1` on the diagonal for a `Raw` direction. On the exact operator, the ARD leg differentiates the unmajorized curvature (`ard_exact_hessian_derivative`, `:6801`). A grep for `clamp` in both functions finds only documentation and majorizer derivations, and no term that differentiates `c_a`.

### The missing term

At first order, with `M = UᵀdAU`,

```text
dλ̃_a = M_aa + dc_a
dc_a  = v_aᵀ dE v_a + 2 Σ_{b≠a} (UᵀEU)_ab M_ab / (λ_a − λ_b)
```

The rotation part `Σ_a λ̃_a d(v_a v_aᵀ)` has off-diagonal coefficient `(λ̃_a − λ̃_b)/(λ_a − λ_b)·M_ab`, which is exactly what `F` supplies. So `DΦ[dA]` covers everything except `Σ_a dc_a v_a v_aᵀ`, and the lane's gradient is short by

```text
Σ_{rows} Σ_{a clamp-basin} g_aa · dc_a,     g_aa = (UᵀGU)_aa
```

It has two pieces:

1. **Explicit.** `g_aa · v_aᵀ (∂E/∂·) v_a`. Both clamp producers are degree-one in their strength (`λ_sparse` on threshold-gate logit slots, `α` on periodic-ARD coordinate slots), so `∂E/∂ρ` is `E` restricted to that coordinate's slots. In θ it is the diagonal `ard_concave_clamp_dt_diagonal` the dense route already uses.
2. **Response.** `2 g_aa Σ_{b≠a} (UᵀEU)_ab M_ab/(λ_a − λ_b)`. This is linear in `dA`, so it folds into the same t–t weight as `deflation_folded_trace_weight`, with the same gap convention for near-degenerate pairs.

A row with no clamp-basin direction contributes nothing.

### The dense route already carries both

On the dense exact-A route, `priced_clamp_adjoint_extras` (`construction_exact_hessian.rs:4232`) chains `clamp_diagonal_derivative` to strength and θ. `exact_hessian_basin_differential` (`:3577`) puts the negative projector's response to `E` into `a_derivative`. So the two routes differ exactly where a row records a clamp basin. The lane's own parity gate on those rows (`threshold_gate_trace_theta_adjoint_matches_from_probes_on_clamp_basin_rows_2333`) compares two channels that share this convention, so it cannot see the gap.

### Acceptance

- Take a state whose exact-A evidence factor records clamp-basin rows (the ARD ladder in `tests_clamp_basin_deflation_2333`).
- Compute a central finite difference, over frozen θ̂ and a fixed discrete stratum, of the lane's `½log|A|` in `log_lambda_sparse`, one ARD precision, and one logit.
- Require it to equal the from-probes channels plus the two pieces above.
- Positive control: the same bar with the pieces removed must fail by more than the tolerance.

**Comment by SauersML** [comment]
## Correction to my comment above: the explicit `dE` leg is not missing for `ρ_sparse`. For that coordinate the lane differentiates the wrong operator.

Found by reading at `47b0134f7`. Nothing here has been run.

### What stands

The explicit clamp leg `g_aa · v_aᵀ (∂E/∂·) v_a` is missing only where a channel differentiates the exact seam. That holds for:

- `ard_log_precision_hessian_trace_from_probes` (`construction_ard.rs:1056`). It contracts `prior.log_precision_curvature(operator)`, which under `ExactObservedInformation` is the unclamped `α·cos κt`.
- `logdet_theta_adjoint_from_probes` (`construction_quasi_laplace.rs:6439`). It contracts `ard_exact_hessian_derivative` and `assignment_prior_hdiag_derivative_entry(.., exact_a = true)`.

In both, `vᵀ (∂A) v` omits `vᵀ (∂E) v`, and the derivation above applies.

### What was wrong

For the sparse coordinate I claimed the same leg was missing. It is not.

- The threshold-gate arm of `assignment_log_strength_hessian_trace_from_probes` (`construction_quasi_laplace.rs:5184`) contracts `hdiag = assignment_prior_log_strength_hdiag_weighted(..)`.
- For `ThresholdGate` that is `ThresholdGateLogitCurvature::psd_majorizer_hess()` (`assignment.rs:1618-1620`), i.e. `∂B/∂ρ_sparse`. The function takes no operator argument.
- So `vᵀ (∂B) v = vᵀ (∂A) v + vᵀ (∂E) v` already carries the clamp term. Adding the explicit leg to this channel would count it twice.

### The actual gap for `ρ_sparse`

On the matrix-free lane the value is `½log|A_cond| + rank_charge` (`outer_objective.rs:1162-1179`, `EvidenceOperator::ExactObservedInformation`). The only sparse log-det trace that lane adds is `assignment_log_strength_hessian_trace_from_probes(.., evidence_operator)` (`construction_exact_hessian.rs:2887`). Under that operator:

- The softmax arm reads `softmax_sparse_curvature_rho_derivative_block(.., operator)`, which is operator-aware.
- The threshold-gate arm reads the majorizer seam whatever the operator, so it contracts `DΦ[∂B/∂ρ_sparse]`.
- `∂A/∂ρ_sparse = ∂B/∂ρ_sparse + ∂ΔC/∂ρ_sparse`. For the gate, `ΔC` on a logit is `negative_hessian_remainder`, nonzero on every switched-on logit and degree one in `λ_sparse`, so `∂ΔC/∂ρ_sparse = ΔC`.
- No production code on the lane contracts it. `threshold_gate_negative_hessian_remainder_weighted` has no caller in `construction_quasi_laplace.rs`, `outer_objective.rs`, `streaming_plan.rs` or `construction_ard.rs`. `exact_stationarity_penalty_derivative_delta_by_flat` is called only on the dense exact-A route (`construction_exact_hessian.rs:2054`, reached from `dense_exact_a_logdet_channels` at `:4335`).

So on a threshold-gate fit taking the lane, the sparse log-det trace is short by `½·[tr(G·diag(ΔC_logit)) − deflation_block_correction(G, diag(ΔC_logit))]` over every row with a switched-on logit. It is not limited to clamp-basin rows. Within the same gradient, the θ-adjoint for those logits reads the exact seam.

### Revised scope for this issue

1. **`ρ_sparse`, threshold gate, exact operator.** Contract `∂B/∂ρ_sparse + ΔC_logit` in the probes arm, from the same producer the dense delta map uses. The existing Daleckii–Krein correction then applies to the combined diagonal.
2. **Clamp-basin price.** The `dc_a` leg for ARD and θ, as derived above. For `ρ_sparse`, after item 1, the clamp price adds `vᵀ(∂E)v = −vᵀ(∂ΔC)v` along clamp-basin directions, plus the response term.
3. **Acceptance.** Central finite difference of the lane's `½ arrow_log_det` over frozen θ̂, on a threshold-gate state with switched-on logits (clamp-basin rows present and absent), against each from-probes channel under `ExactObservedInformation`.
